docs/GETTING_STARTED.md

```diff
@@ -0,0 +1,144 @@
+![immutadot logo](https://raw.githubusercontent.com/Zenika/immutadot/master/misc/otter.svg?sanitize=true)
+===
+
+# Getting started
+
+This guide will give you an overview of immutad●t features.
+
+Feel free to try the following examples with [runkit](https://npm.runkit.com/immutadot).
+
+## Installation
+
+immutad●t is available on [npm repository](https://www.npmjs.com/package/immutadot).
+
+using yarn:
+
+```shell
+$ yarn add immutadot
+```
+
+using npm:
+
+```shell
+$ npm install immutadot
+```
+
+or you can directly download [sources](https://github.com/Zenika/immutadot/releases).
+
+## Setting nested properties
+
+[ES2015+ destructuring](https://mdn.io/destructuring) provides you all necessary tools to keep nested structures immutable. The [spread operator](https://mdn.io/Spread_operator) is a succinct syntax create new arrays and objects using existing ones.
+
+```js
+const otter = {
+  family: 'Mustelidae',
+  order: 'Carnivora'
+}
+const updatedOtter = {
+  ...otter,
+  classification: 'Mammalia'
+}
+```
+
+With nested structures this syntax becomes more tedious to write, and harder to read:
+
+```js
+const animals = {
+  weasels: {
+    badger: {
+      vernacularName: 'badger',
+      scientificName: 'Meles meles'
+    },
+    otter: {
+      vernacularName: 'otter',
+    }
+  }
+}
+
+const newAnimals = {
+  ...animals,
+  weasels: {
+    ...animals.weasels,
+    otter: {
+      ...animals.weasels.otter,
+      scientificName: 'Lutrinae'
+    }
+  }
+}
+```
+
+This can be done nicely with immutadot's [`set()`](https://zenika.github.io/immutadot/immutadot/1.0/core.html#.set) core utility:
+
+```js
+const updatedAnimals = set(animals, 'animals.weasels[1].scientificName', 'Lutrinae')
+```
+
+It seems much more clear, no?
+
+## Basic array operations
+
+Assume that our badger object have a property sharingName that is an array of other species with the common name of badger but our favorite one is missing.
+
+We are pretty fond of honey badger aka ratel. Let's add it in our weasels array with [`push()`](https://zenika.github.io/immutadot/immutadot/1.0/array.html#.push) function.
+
+```js
+const updatedAnimals = push(animals, 'animals.weasels.badger.sharingName', 'Honey badger')
+```
+
+There are a few more operations you can find out in our [documentation's array section](https://zenika.github.io/immutadot/immutadot/1.0/array.html).
+
```

Comment:
I think this would be a good place for a section on slice notation, because this is another kind of "basic array operation".
For example capitalize all the commonNames.